File: Perl/pe7.pl
sub isPrime
{
	$n = shift;
	for $p (2..($n - 1))
	{
		if($n % $p == 0)
		{
			return 0;
		}
	}
	return 1;
}

loop:
$i++;
if(isPrime($i)) 
{
	push(@primes, $i);
}
$ele = @primes;
if($ele <= 10001)
{
	goto loop;
}
print @primes[-1];

File: C++/PE7.cpp
#include<iostream>

using namespace std;

int isPrime(int i)
{
	for(int j = 2; j <= (i - 1); j++)
	{
		if(i % j == 0)
		{
			return 0;
		}
	}
	return 1;
}

int main()
{
	int i = 1;
	int n = 1;
	while(1)
	{
		i++;
		if(isPrime(i))
		{
			n++;
			if(n == 10002)
			{
				break;
			}
		}
	}
	cout << i << endl;
	system("pause");
	return 0;
}

